posthog/src/main/java/com/posthog/internal/PostHogQueue.kt

```diff
@@ -48,60 +48,89 @@ internal class PostHogQueue(
 
     private val delay: Long get() = (config.flushIntervalSeconds * 1000).toLong()
 
-    override fun add(event: PostHogEvent) {
-        executor.executeSafely {
-            var removeFirst = false
-            if (deque.size >= config.maxQueueSize) {
-                removeFirst = true
+    private fun addEventSync(event: PostHogEvent): Boolean {
+        storagePrefix?.let {
+            val dir = File(it, config.apiKey)
+
+            if (!dirCreated) {
+                dir.mkdirs()
+                dirCreated = true
             }
 
-            if (removeFirst) {
-                try {
-                    val first: File
-                    synchronized(dequeLock) {
-                        first = deque.removeFirst()
-                    }
-                    first.deleteSafely(config)
-                    config.logger.log("Queue is full, the oldest event ${first.name} is dropped.")
-                } catch (ignored: NoSuchElementException) {
+            val uuid = event.uuid ?: TimeBasedEpochGenerator.generate()
+            val file = File(dir, "$uuid.event")
+            synchronized(dequeLock) {
+                deque.add(file)
+            }
+
+            try {
+                val os = config.encryption?.encrypt(file.outputStream()) ?: file.outputStream()
+                os.use { theOutputStream ->
+                    config.serializer.serialize(event, theOutputStream.writer().buffered())
                 }
+                config.logger.log("Queued Event ${event.event}: ${file.name}.")
+
+                return true
+            } catch (e: Throwable) {
+                config.logger.log("Event ${event.event}: ${file.name} failed to parse: $e.")
+
+                // if for some reason the file failed to serialize, lets delete it
+                file.deleteSafely(config)
             }
 
-            storagePrefix?.let {
-                val dir = File(it, config.apiKey)
+            return false
+        }
 
-                if (!dirCreated) {
-                    dir.mkdirs()
-                    dirCreated = true
-                }
+        // if there's no storagePrefix, we assume it failed
+        return true
+    }
+
+    private fun removeEventSync() {
+        var removeFirst = false
+        if (deque.size >= config.maxQueueSize) {
+            removeFirst = true
+        }
 
-                val uuid = event.uuid ?: TimeBasedEpochGenerator.generate()
-                val file = File(dir, "$uuid.event")
+        if (removeFirst) {
```

Comment:
Why not just this?

```
if (deque.size >= config.maxQueueSize)
```

Reply:
i was reusing removeFirst before but not anymore, good catch, will remove